persistence/src/main/java/edu/unc/lib/dl/persist/services/destroy/DestroyObjectsJob.java

```diff
@@ -0,0 +1,154 @@
+public class DestroyObjectsJob implements Runnable {
+    private static final Timer timer = TimerFactory.createTimerForClass(DestroyObjectsJob.class);
+
+    private List<PID> objsToDestroy;
+    private AgentPrincipals agent;
+    private RepositoryObjectFactory repoObjFactory;
+    private RepositoryObjectLoader repoObjLoader;
+    private TransactionManager txManager;
+    private DestroyProxyService proxyService;
+    private ObjectPathFactory pathFactory;
+
+    public DestroyObjectsJob(AgentPrincipals agent, List<PID> objsToDestroy) {
+        this.objsToDestroy = objsToDestroy;
+    }
+
+    @Override
+    public void run() {
+        FedoraTransaction tx = txManager.startTransaction();
+        try (Timer.Context context = timer.time()) {
+            // convert each destroyed obj to a tombstone
+            for (PID pid : objsToDestroy) {
+                // remove containment relation from obj's parent
+                proxyService.destroyProxy(pid);
+                RepositoryObject repoObj = repoObjLoader.getRepositoryObject(pid);
+                // purge tree with repoObj as root from repository
+                destroyTree(repoObj);
+           }
+        } catch (Exception e) {
+             tx.cancel(e);
+        } finally {
+             tx.close();
+        }
+    }
+
+    private void destroyTree(RepositoryObject rootOfTree) throws FedoraException {
+        if (rootOfTree instanceof ContentContainerObject) {
+            ContentContainerObject container = (ContentContainerObject) rootOfTree;
+            List<ContentObject> members = container.getMembers();
+            for (ContentObject member : members) {
+                destroyTree(member);
+            }
+        } else if (rootOfTree instanceof FileObject) {
+            FileObject file = (FileObject) rootOfTree;
+            List<BinaryObject> binaries = file.getBinaryObjects();
+            for (BinaryObject binary : binaries) {
+                destroyTree(binary);
+            }
+        }
+        // destroy root of tree
+        Model stoneModel = rootOfTree.getModel();
+        stoneModel = convertModelToTombstone(rootOfTree);
+        repoObjFactory.createObject(rootOfTree.getUri(), stoneModel);
+
+        //add premis event to tombstone
+        rootOfTree.getPremisLog().buildEvent(Premis.Deletion)
+        .addImplementorAgent(agent.getUsernameUri())
+        .addEventDetail("Item deleted from repository and replaced by tombstone")
+        .write();
+
+        // persist model
+        repoObjFactory.createObject(rootOfTree.getUri(), stoneModel);
+    }
+
+    private Model convertModelToTombstone(RepositoryObject destroyedObj) {
+        Model model = destroyedObj.getModel();
+        Resource resc = destroyedObj.getResource();
+
+        if (destroyedObj instanceof BinaryObject) {
+            // whitelisted properties for tombstone model of binaries
+            Selector selectEbucoreFilename = new SimpleSelector(resc, Ebucore.filename, (Object) null);
```

Comment:
I think this approach will work. Looking at the jena documents, a slight variation would be to extend SimpleSelector with a custom .selects method which tests against all of the requested predicates. Its probably slightly more performent, but may improve readability. This is based off the second to last example in the "Querying a Model" section here https://jena.apache.org/tutorials/rdf_api.html#ch-Querying%20a%20Model
```
// Or MultiPropertySelector
class BinaryPropertySelector extends SimpleSelector {
          List<Property> permittedPredicates = // List of predicates

          public BinaryPropertySelector(Resource subject) {
              // call super constructor
           }

          @Override
          public boolean selects(Statement s) {
              return (subject == null   || s.getSubject().equals(subject))
                  && (predicate == null || /* Compare against a list of predicates */)
                  && (object == null    || s.getObject().equals(object)) ;
          }
     }
}
```
And to use:
`StmtIterator iter = model.listStatements(new BinaryPropertySelector(resc));`

Reply:
This might make it a bit easier to have a predefined list of properties as a constant